app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -0,0 +1,123 @@
+    }
+
+    // TODO: set state accurately, choose correct subject, set initial state, etc.
+    val accountStateDontUseMeYet = BehaviorSubject.create<AccountState>()
+
+    init {
+        accountManager.register(FirefoxAccountObserver())
+        accountManager.registerForDeviceEvents(FirefoxDeviceEventsObserver(), ProcessLifecycleOwner.get(),
+            autoPause = false /* Avoid pausing even when the app is backgrounded. */)
+
+        @Suppress("DeferredResultUnused") // No value is returned & we don't need to wait for this to complete.
+        accountManager.initAsync() // If user is already logged in, the appropriate observers will be triggered.
+    }
+
+    /**
+     * Notifies the FxA library that login is starting: callers should generally call [FxaLoginUseCase.beginLogin]
+     * instead of this method.
+     */
+    fun beginLoginInternalAsync(): Deferred<String?> {
+        return accountManager.beginAuthenticationAsync()
+    }
+
+    private inner class FirefoxAccountObserver : AccountObserver {
+        override fun onAuthenticated(account: OAuthAccount) {
+            Log.d(LOGTAG, "onAuthenticated")
+            // todo: is this correct?
+            val nextState = if (accountManager.accountProfile() != null) AUTHENTICATED else AUTHENTICATED_NO_PROFILE
+            accountStateDontUseMeYet.onNext(nextState)
+        }
+
+        override fun onAuthenticationProblems() {
+            Log.d(LOGTAG, "onAuthenticationProblems")
+            accountStateDontUseMeYet.onNext(NEEDS_REAUTHENTICATION)
+        }
+
+        override fun onError(error: Exception) {
+            // This is for internal errors in the sync library and is an unexpected state.
+            Log.d(LOGTAG, "onError")
+        }
+
+        override fun onLoggedOut() {
+            Log.d(LOGTAG, "onLoggedOut")
+            accountStateDontUseMeYet.onNext(NOT_AUTHENTICATED)
+        }
+
+        override fun onProfileUpdated(profile: Profile) {
+            Log.d(LOGTAG, "onProfileUpdated")
+            // todo: update state.
+        }
+    }
+
+    private inner class FirefoxDeviceEventsObserver : DeviceEventsObserver {
+        override fun onEvents(events: List<DeviceEvent>) {
+            Log.d(LOGTAG, "received device events: $events")
+        }
+    }
+
+    companion object {
+        // TODO #2506: use production FFTV client ID (this ID is from an FxA sample app).
+        private const val CLIENT_ID = "a2270f727f45f648"
+        const val REDIRECT_URI = "https://accounts.firefox.com/oauth/success/$CLIENT_ID"
+
+        private fun newInstanceDefaultAccountManager(context: Context): FxaAccountManager {
+            return FxaAccountManager(
+                context,
+                Config.release(CLIENT_ID, REDIRECT_URI),
+                applicationScopes = arrayOf("profile", "https://identity.mozilla.com/apps/oldsync"), // todo: proper vals?
+                deviceTuple = DeviceTuple(
+                    name = "Firefox for Fire TV",
```

Comment:
Nit: R.string.app_name_extended_fire

Reply:
This will be resolved in #2516: we probably want to do more than use `R.string.app_name...`